tests/test_config.py

```diff
@@ -29,10 +28,22 @@ def autouse_config_tmpdir(config_tmpdir):
 
 def cfg(f=ALL, topdir=None):
     # Load a fresh configuration object at the given level, and return it.
-    cp = configparser.ConfigParser(allow_no_value=True)
-    config.read_config(configfile=f, config=cp, topdir=topdir)
+    cp = config.configparser.ConfigParser(allow_no_value=True)
+    cp.read(config._gather_configs(f, topdir), encoding='utf-8')
     return cp
 
+def update_config(*args, **kwargs):
+    # For testing purposes it's easiest to use the deprecated functions
+    # but make these calls explicit.
+    with pytest.deprecated_call():
+        return config.update_config(*args, **kwargs)
+
+def delete_config(*args, **kwargs):
```

Comment:
Warning: this review comment will be hard to find because that line  is changed in a later commit in the PR and Github doesn't like the multi-commit, force-push workflow.

There is already an API named `delete_config()`, so please use a different name for internal symbols. Example:
```suggestion
def delete_testcfg(*args, **kwargs):
```

Same for `update_config` above.

Sincere condolences for the ripple effect on consecutive commits :-(

Generally speaking, please choose less "API-looking" names for internal symbols. Pretty much everything here is called `something_config` and it's incredibly hard to keep track of what is: 1) Builtin Python (e.g. `configparser`) 2) deprecated west API 3) current west API 4) private west objects abused for "white box" testing 5) local/private test symbols,....

It's never going to be perfect and please do not cause massive git churn but brand new symbols is where progress can and should be made.

This is a cleanup PR so it should bring more clarity.

Reply:
Yeah, I tried to do it in some reasonable steps, but I think it can make it harder to follow.
I'll come back to this PR, but there's no urgency here, as I didn't intend on getting into 1.3.

Reply:
Renamed as suggested, which is clearer.